From: Andrew Morton <akpm@linux-foundation.org>
Date: Mon, 3 Aug 2026 17:44:48 -0700
Subject: [PATCH] 
 mm-page_alloc-only-update-lowmem_reserve_ratio-on-sysctl-write-fix

use ARRAY_SIZE to dimension ratio[]

Cc: Brendan Jackman <jackmanb@google.com>
Cc: Jianlin Shi <shijianlin11@foxmail.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@kernel.org>
Cc: Zi Yan <ziy@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/page_alloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index 99e33f27a3ffb..408364ff2c93a 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -6926,7 +6926,7 @@ static int lowmem_reserve_ratio_sysctl_handler(const struct ctl_table *table,
 		int write, void *buffer, size_t *length, loff_t *ppos)
 {
 	struct ctl_table tmp = *table;
-	int ratio[MAX_NR_ZONES];
+	int ratio[ARRAY_SIZE(sysctl_lowmem_reserve_ratio)];
 	int rc;
 
 	if (!write)
-- 
2.47.3

